src/regdesk/label_builder.js

```diff
@@ -0,0 +1,217 @@
+
+    // TODO: Figure out a better way to do this?
+    const font = new FontFace("twemoji", 'url(fonts/TwitterColorEmoji-SVGinOT.woff2)', {
+        unicodeRange: emojiUnicodeRange,
+    });
+    doc.fonts.add(await font.load());
+  }
+
+  /**
+   * Add this builder's details to a label.
+   * @param {LabelEpl} label - The label to add the print details to.
+   * @param {number} heightPadding - Dots to add to the top of the label as padding.
+   * @param {number} widthPadding - Dots to divide btween either side of the label as padding.
+   */
+  addToLabel(label, heightPadding = 10, widthPadding = 20) {
+    // Safety buffer around the edges of the printable area.
+    // Make sure the width ends up on an byte boundary!
+    let widthRemainder = (label.labelWidthDots - widthPadding) % 8;
+    let widthOffset = widthPadding + widthRemainder;
+    let canvasData = this.renderToImageData(
+      label.labelWidthDots - widthOffset,
+      label.labelHeightDots - heightPadding);
+
+    label.setOffset(widthOffset / 2).addImage(canvasData);
```

Comment:
It is, I mentioned [here](https://github.com/anthroarts/RegExtension/pull/10#discussion_r775310971) that I'd write an extension method that would be a neat(-er) way of handling this. I'm uncertain if the JS equivalent of mucking with object prototypes would be an appropriate solution here.

Basically, add a method onto the LabelEpl class that takes in the LabelBuilder as a parameter and adds the contents of the LabelBuilder to the LabelEpl. The relationship would be far more clear in that case. 

As for this particular method, since the information we need to modify the label is also present within the label, I presumed that it'd be easier to just work with the label directly instead of expecting it to be managed externally. 

My spidey sense goes off when I tend to pick some bits of an object out, pass them to a function, take the results, and then put them back into the same object. If the state is so closely related, why not just hand the state off entirely? 

I can move the label modification out.

Reply:
Your spidey senses are right. That definitely weird. But its the whole "read some, write some" that's wrong in the first place. That is, an object shouldn't be used to mutate itself. Get into all kinds of hard to test/debug scenarios with that.

So you're right, my ultimate objection is refactor it /all/ to not do that. This is, in my mind, the first step. Decouple the Label from the LabelBuilder.

Reply:
I've swapped it around and I do kinda feel that the complexity is now smeared across to the call site. I'm uncertain if the tradeoff there is worth it.

Really jonesing for my extension methods right now, they're one of the best ways to handle this kind of glue between classes..

Reply:
Thanks GitHub for dropping the websocket and not showing me a reply..

Reply:
I think this is a lot better. The label is entirely self contained inside printLabelBuilder method. LabelBuilder doesn't need to know anything about it. The client constructs a LabelBuilder, passes it to PrinterManager without any back and forth. No mutation of variables. I honestly think this is the best and a super clean solution!